Mod2/Lessons/Week4/LINQ.md

```diff
@@ -1,15 +1,199 @@
+
+One of the more common use cases for combining LINQ methods is ordering a filtered list.  
+
+> With a partner, review the code below and answer the following questions:
+> 1. How many methods are being executed?
+> 2. Why is this statement on multiple lines?
+> 3. Could we re-order this statement?
+> 4. 🌶️What does `Distinct()` do?
+>
+> Be ready to share out!
+
+```c#
+var orderedCNames = names.Where(name => name[0] == 'C')
+    .OrderBy(name)
+    .Distinct();
+```
+
+** Instructor Note ** Use the last question about `Distinct` to introduce the [LINQ method docs](https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable.distinct?view=net-7.0)
+
+### Researching
+There are too many LINQ methods to cover in one lesson.  LINQ is a tool that will encourage a lot of research as you learn about new (to you) methods.  Many tasks you need to complete with a collection of data can be done with a LINQ method, so it is good to familiarize yourself with some of the documentation.
+
+> With a partner, research the following LINQ methods.  Try to use each method to answer the song-related questions included:
+> 1. All - Do all of the songs have a playCount higher than 10?
+> 2. Sum - If I made a playlist with all of the songs, how long would the playlist be?
+> 2. First - What is the first song in the list by Talking Heads?
+> 4. 🌶️ How long would a playlist just for Madonna songs be?
+
+
+## Query Syntax
+
+To this point, we have used **method syntax** for creating LINQ queries.  This is the preferred way to create LINQ queries, but it is not the _only_ way.  As you are researching LINQ, you may come across **query syntax**.  You do not need to be proficient at query syntax, but it is good to recognize it when you see it on documentation, or blog posts. 
+
+Take a look at the code below:
+
+```c#
+class QueryVMethodSyntax
+{
+    static void Main()
+    {
+        int[] numbers = { 5, 10, 8, 3, 6, 12};
+
+        //Query syntax:
+        var orderedEvenNumsQuery =
+            from num in numbers
+            where num % 2 == 0
+            orderby num
+            select num;
+
+        //Method syntax:
+        var OrderedEvenNumsMethod = numbers.Where(num => num % 2 == 0).OrderBy(n => n);
+
+        Console.Write("Query Syntax: ");
+        foreach (int i in orderedEvenNumsQuery)
+        {
+            Console.Write(i + " ");
+        }
+
+        Console.WriteLine();
+        Console.Write("Method Syntax: ");
+        foreach (int i in orderedEvenNumsMethod)
+        {
+            Console.Write(i + " ");
+        }
+    }
+}
+/*
+    Output:
+    Query Syntax: 6 8 10 12
+    Method Syntax: 6 8 10 12
+ */
+ ```
+
+ Query and Method syntax both accomplish the same thing, but Method syntax is a bit easier to write and read.  So, nearly all of the time, we will ask you to use Method syntax when using LINQ.
```

Comment:
I would maybe change this wording to say that we will never ask you to do query syntax. I assume that's the plan right, that she shouldn't think they need to learn query syntax in this course.

Reply:
Yep - rewording it, but going to shy away from `never` 😆